### PR TITLE
add omit empty to metadata

### PR DESCRIPTION
**Pull request by ramaraochavali**
When we display /debug/connections we do not display metadata

- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [X] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

**Review by howardjohn — APPROVED** [review]

File: pilot/pkg/xds/debug.go [hunk 1]
```diff
@@ -97,7 +97,7 @@ type AdsClient struct {
 	ConnectionID string              `json:"connectionId"`
 	ConnectedAt  time.Time           `json:"connectedAt"`
 	PeerAddress  string              `json:"address"`
-	Metadata     *model.NodeMetadata `json:"metadata"`
+	Metadata     *model.NodeMetadata `json:"metadata,omitempty"`
```
Comment (hzxuzhonghu) [inline pilot/pkg/xds/debug.go:100]: When can this field be empty

Comment (ramaraochavali) [reply pilot/pkg/xds/debug.go:100]: when /connections is displayed https://github.com/istio/istio/blob/master/pilot/pkg/xds/debug.go#L511

**Pull request merged**